src/main/java/com/onionshop/managers/DrawingManager.java

```diff
@@ -39,47 +27,10 @@ public DrawingManager(Tool selectedTool, Colour selectedColour, Project curProje
      * @return A list of pixels to update. This list only contains the pixel location, not the color data.
      */
     public int[][] updateCanvasAfterStroke(int x, int y) {
-        int[][] pixelsToReturn = this.currentlySelectedTool.draw(currentProject, currentlySelectedColour, x, y);
+        int[][] pixelsToReturn = ToolStateManager.getInstance().getCurrentToolState().draw(
```

Comment:
Interesting, I think this makes sense, but Im just highlighting cuz I wanna ask the TA about it, if we can make this cleaner somehow, idk